src/AutoSaveStatus/AutoSaveStatusProvider.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { PropsWithChildren, useCallback, useEffect, useRef, useState } from "react";
+
+export enum AutoSaveStatus {
+  IDLE = "idle",
+  SAVING = "saving",
+  DONE = "done",
+  ERROR = "error",
+}
+
+export interface AutoSaveStatusContextType {
+  status: AutoSaveStatus;
+  /** Resets status to IDLE, particularly useful if "Error" or "Done" is stale */
+  resetStatus: VoidFunction;
+  errors: unknown[];
+  /** Notifies AutoSaveContext that a request is in-flight */
+  triggerAutoSave: VoidFunction;
+  /** Notifies AutoSaveContext that a request has settled, optionally taking an error */
+  resolveAutoSave: (error?: unknown) => void;
+}
+
+export const AutoSaveStatusContext = React.createContext<AutoSaveStatusContextType>({
+  status: AutoSaveStatus.IDLE,
+  resetStatus() {},
+  errors: [],
+  triggerAutoSave() {},
+  resolveAutoSave() {},
+});
+
+type AutoSaveStatusProviderProps = PropsWithChildren<{
+  /** After a successful save, reset Status back to `Idle` after this many milliseconds */
+  resetToIdleTimeout?: number;
```

Comment:
Nice, makes sense, to avoid a flashy save / "it's gone!" / save / "it's gone!" experience.